modules/explorer-api/src/main/scala/org/ergoplatform/explorer/http/api/v1/shared/MempoolProps.scala

```diff
@@ -43,6 +47,29 @@ object MempoolProps {
 
     import repo._
 
+    def getUnconfirmedBalanceByAddress(
+      address: Address,
+      confirmedBalance: Balance
+    ): F[Balance] = {
+      val ergoTree = addressToErgoTreeNewtype(address)
+      txs
+        .countByErgoTree(ergoTree.value)
+        .flatMap { total =>
+          txs
+            .streamRelatedToErgoTree(ergoTree, 0, Int.MaxValue)
+            .chunkN(settings.chunkSize)
+            .through(mkTransaction)
+            .to[List]
+            .map { poolItems =>
+              total match {
```

Comment:
Why branching on `total` is so far away? You don't need to do all those things if total is 0
```
          txs
             .streamRelatedToErgoTree(ergoTree, 0, Int.MaxValue)
             .chunkN(settings.chunkSize)
             .through(mkTransaction)
             .to[List]
```